worker/processors/hooks/index.js

```diff
@@ -0,0 +1,163 @@
+const enrichData = require('./enrichEventData');
+
+module.exports = async (job) => {
+  sails.log.debug('[Worker] Got a `hooks` job', job.data);
+  return handle(job.data);
+};
+
+
+async function handle(data) {
+  const eventData = data.data;
+  const eventType = data.type;
+
+  // First we handle any 'logLine' events.
+  if (eventType === 'logLine') {
+    const serverLogLineHooks = await CustomHook.find({
+      server: eventData.server.id,
+      event: 'logLine'
+    });
+
+    if (!serverLogLineHooks) {
+      serverLogLineHooks = [];
+    }
+
+    for (const serverLogLineHook of serverLogLineHooks) {
+      const stringFound = checkLogLine(`${eventData.time} ${eventData.date} ${eventData.msg}`, serverLogLineHook);
+
+      if (stringFound) {
+        sails.log.debug(`Found the string! Executing hook ${serverLogLineHook.id}`);
+        const isNotOnCooldown = await handleCooldown(serverLogLineHook);
+        if (isNotOnCooldown) {
+          const variables = await getHookVariables(serverLogLineHook.id);
+          serverLogLineHook.variables = variables;
+          await executeLogLineHook(eventData, serverLogLineHook, eventData.server.id);
+        }
+      }
+    }
+    return;
+  }
+
+  // Handle any built-in events
+  const configuredHooks = await CustomHook.find({
+    server: eventData.server.id,
+    event: eventType
+  });
+
+  for (const hookToExec of configuredHooks) {
+    const isNotOnCooldown = await handleCooldown(hookToExec);
+    const stringFound = checkLogLine(`${eventData.time} ${eventData.date} ${eventData.msg}`, hookToExec);
+    if (stringFound) {
+      if (isNotOnCooldown) {
+        const variables = await getHookVariables(hookToExec.id);
+        hookToExec.variables = variables;
+        await executeHook(eventData, hookToExec, eventData.server.id);
+      }
+    }
+  }
+}
+
+
+
+async function executeHook(eventData, hookToExec, serverId) {
+  let server = await SdtdServer.findOne(serverId);
+  eventData.server = server;
+  eventData = await enrichData(eventData);
+  eventData.custom = getVariablesValues(hookToExec.variables, eventData.msg);
+  let results = await sails.helpers.sdtd.executeCustomCmd(server, hookToExec.commandsToExecute.split(';'), eventData);
+  await saveResultsToRedis(hookToExec.id, results);
+  sails.log.debug(`Executed a custom hook for server ${serverId}`, {
+    hook: hookToExec,
+    event: eventData,
+    results: results
+  });
+}
+
+async function executeLogLineHook(eventData, hookToExec, serverId) {
+
+  // Ignore log messages created by the web API. This is the prevent spam
+  if (eventData.msg.includes('WebCommandResult_for_')) {
+    return;
+  }
+
+  let server = await SdtdServer.findOne(serverId);
+  eventData.server = server;
+  eventData = await enrichData(eventData);
+  eventData.custom = getVariablesValues(hookToExec.variables, eventData.msg);
+  let results = await sails.helpers.sdtd.executeCustomCmd(server, hookToExec.commandsToExecute.split(';'), eventData);
```

Comment:
sails.helpers.sdtd.executeCustomCmd

soooo, https://media.discordapp.net/attachments/702653693115498496/773608441171935262/unknown.png

Which is right?

Reply:
![image](https://user-images.githubusercontent.com/110087/98336069-0baa7d00-1fbb-11eb-932a-5e96e6be95ba.png)

So... I think this needs test coverage

Reply:
I don't follow how this is relevant here.

`sails.helpers.sdtdApi` 
vs
`sails.helpers.sdtd`

right? What am I missing?